src/extensions/anvil/algorithm.ts

```diff
@@ -83,6 +84,7 @@ async function initialize(runtime: RuntimeContext, structure: Structure, props:
 
     let offsets = new Array<number>(elementCount);
```

Comment:
If you are using Array (as opposed to a typed arrays) you can .push elements without performance penalty and then you don't need to slice them later which creates a new copy. Alternatively you can use typed arrays and then get correctly sized ones without copying by using TypedArray.buffer like new Uint32Array(ta.buffer, 0, n).

Reply:
Got it. Moved to `.push` and removed `.slice` in 31f4803.